src/app/server/Server.cpp

```diff
@@ -460,7 +402,65 @@ CHIP_ERROR OpenDefaultPairingWindow(ResetFabrics resetFabrics, chip::PairingWind
     VerifyOrReturnError(fabricInfo != nullptr, CHIP_ERROR_NO_MEMORY);
     gNextAvailableFabricIndex++;
 
-    return gRendezvousServer.WaitForPairing(std::move(params), &gExchangeMgr, &gTransports, &gSessions, fabricInfo);
+    ReturnErrorOnFailure(gRendezvousServer.WaitForPairing(
+        std::move(params), kSpake2p_Iteration_Count,
+        ByteSpan(reinterpret_cast<const unsigned char *>(kSpake2pKeyExchangeSalt), strlen(kSpake2pKeyExchangeSalt)), 0,
+        &gExchangeMgr, &gTransports, &gSessions, fabricInfo));
+
+    if (commissioningTimeoutSeconds != kNoCommissioningTimeout)
+    {
+        ReturnErrorOnFailure(
+            DeviceLayer::SystemLayer.StartTimer(commissioningTimeoutSeconds * 1000, HandlePairingWindowTimeout, nullptr));
+    }
+
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR OpenPairingWindowUsingVerifier(uint16_t commissioningTimeoutSeconds, uint16_t discriminator, PASEVerifier & verifier,
+                                          uint32_t iterations, ByteSpan salt, uint16_t passcodeID)
+{
+    RendezvousParameters params;
+
+    ReturnErrorOnFailure(gDeviceDiscriminatorCache.UpdateDiscriminator(discriminator));
+
+#if CONFIG_NETWORK_LAYER_BLE
+    params.SetPASEVerifier(verifier)
+        .SetBleLayer(DeviceLayer::ConnectivityMgr().GetBleLayer())
+        .SetPeerAddress(Transport::PeerAddress::BLE())
+        .SetAdvertisementDelegate(&gAdvDelegate);
+#else
```

Comment:
The command to open the commissioning window is received over a CASE channel. Do you mean that BLE should not be used for opening the ECM pairing window?

Reply:
Yes

Reply:
Or rather, if opening CASE over BLE is possible, I think that violates unicast communication rules for PASE/CASE. This can be handled by a TODO later but I was under the impression that CASE sessions could only be opened over IP transports

Reply:
The way I would imagine this working is that the command to open the commissioning window comes in over CASE, and then the device starts advertising via both mdns and BLE (if it supports BLE).  Is that not what the code was foing @tcarmelveilleux @pan-apple

Reply:
That's correct. The command to open the commissioning window comes over a secure session established via CASE. The previous code was advertising using BLE if the device supported BLE. Now, it uses IP. My understanding is that when the window is opened, the device already has IP address assigned, so it can do `onnetwork` provisioning.